lib/control-connection.js

```diff
@@ -95,10 +95,12 @@ ControlConnection.prototype.init = function (callback) {
           var portSeparatorIndex = name.lastIndexOf(']:');
           return addHost(name.substr(1, portSeparatorIndex - 1), name.substr(portSeparatorIndex + 2), eachNext);
         }
-        if (name.indexOf('.') > 0 && name.indexOf(':') > 0) {
+        if (name.indexOf(':') > 0) {
```

Comment:
@tolbertam You're spot on with my ISP (Cox) resolving the DNS so that it can show a page of adverts if you browse to anything that doesn't resolve. They even return a 200 status:

```
$ curl -I not1.existent-host
HTTP/1.1 200 OK
Server: nginx
Date: Mon, 15 May 2017 22:14:57 GMT
Content-Type: text/html; charset=utf-8
Connection: close
Expires: Mon, 15 May 2017 22:24:57 GMT
Cache-Control: max-age=600
```